heartbeat/monitors/active/http/http_test.go

```diff
@@ -479,25 +479,27 @@ func TestRedirect(t *testing.T) {
 	sched, _ := schedule.Parse("@every 1s")
 	job := wrappers.WrapCommon(jobs, "test", "", "http", sched, time.Duration(0))[0]
 
-	event := &beat.Event{}
-	_, err = job(event)
-	require.NoError(t, err)
-
-	testslike.Test(
-		t,
-		lookslike.Strict(lookslike.Compose(
-			hbtest.BaseChecks("", "up", "http"),
-			hbtest.SummaryChecks(1, 0),
-			minimalRespondingHTTPChecks(testURL, 200),
-			lookslike.MustCompile(map[string]interface{}{
-				"http.redirects": []string{
-					server.URL + redirectingPaths["/redirect_one"],
-					server.URL + redirectingPaths["/redirect_two"],
-				},
-			}),
-		)),
-		event.Fields,
-	)
+	for i := 0; i < 10; i++ {
```

Comment:
We now run the test here multiple times to catch the original issue

Reply:
Please add a comment explaining why we have to do so. Maybe with link to issue. On first sight this loop looks like a great thing to optimize away :)

Reply:
What is the effect of running this multiple times? Is it some flakiness or does something change in the setup? Can we consutrct some routing creating a huge number of redirects in the fake http server used for testing?

Reply:
Added the comment. The effect is that if you go through say two redirects, you'll see a list of two redirects the first time the check runs, 4 the second time it runs, 6 the time after that, etc. We're using a `*[]string` over and over and not resetting it between runs.